File: hosung/leetcode/841. Keys and Rooms.py
class Solution(object):
    def canVisitAllRooms(self, rooms):
        st = [0]
        ch = [0]*len(rooms)

        while st:
            l = st.pop()
            ch[l] = 1

            for i in rooms[l]:
                if ch[i] ==0:
                    st.append(i)

        for i in ch:
            if i == 0:
                return False
        else:
            return True
                    

File: taejoong/leetcode/841_KeysandRooms.java
class Solution {
    public boolean canVisitAllRooms(List<List<Integer>> rooms) {

        Stack<Integer> stack = new Stack<>();
        boolean[] visited = new boolean[rooms.size()];

        stack.push(0);

        while (!stack.isEmpty()) {

            int room = stack.pop();

            if (visited[room]) {
                continue;
            }

            visited[room] = true;

            for (int key : rooms.get(room)) {

                stack.push(key);
            }
        }

        for (boolean v : visited) {

            if (!v) {
                return false;
            }
        }

        return true;
    }
}